devtools/conda-envs/test_env.yaml

```diff
@@ -4,7 +4,7 @@ channels:
 dependencies:
     # Base depends
   - cython
-  - hdf5 <=1.14.0  # Macos has problem with newer releases (1.14.x) to date
+    #- hdf5 <=1.14.0  # Macos has problem with newer releases (1.14.x) to date
```

Comment:
I do not fully remember why we had to pin the version at the time, but this was probably something that was not fixed upstream (with the python bindings/wrappers). Good to see it's working without pinning!!